File: testdata/jennies/rawtypes/struct_with_optional_fields/JavaRawTypes/struct_optional_fields/StructOptionalFieldsSomeStructFieldAnonymousStruct.java
package struct_optional_fields;

import java.util.Objects;

public class StructOptionalFieldsSomeStructFieldAnonymousStruct {
    public Object fieldAny;
    public StructOptionalFieldsSomeStructFieldAnonymousStruct() {
        this.fieldAny = new Object();
    }
    public StructOptionalFieldsSomeStructFieldAnonymousStruct(Object fieldAny) {
        this.fieldAny = fieldAny;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof StructOptionalFieldsSomeStructFieldAnonymousStruct)) return false;
        StructOptionalFieldsSomeStructFieldAnonymousStruct o = (StructOptionalFieldsSomeStructFieldAnonymousStruct) other;
        if (!Objects.equals(this.fieldAny, o.fieldAny)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.fieldAny);
    }
}

File: testdata/jennies/rawtypes/struct_with_optional_fields/PHPRawTypes/src/StructOptionalFields/StructOptionalFieldsSomeStructFieldAnonymousStruct.php
<?php

namespace Grafana\Foundation\StructOptionalFields;

class StructOptionalFieldsSomeStructFieldAnonymousStruct implements \JsonSerializable
{
    /**
     * @var mixed
     */
    public $fieldAny;

    /**
     * @param mixed|null $fieldAny
     */
    public function __construct( $fieldAny = null)
    {
        $this->fieldAny = $fieldAny ?: null;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->fieldAny !== $other->fieldAny) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{FieldAny?: mixed} $inputData */
        $data = $inputData;
        return new self(
            fieldAny: $data["FieldAny"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->FieldAny = $this->fieldAny;
        return $data;
    }
}
